### ISSUE TITLE
insertAdjacentHTML is broken

### ISSUE THREAD

**Issue by annevk** [w3c/DOM-Parsing#4]
It cannot use "insert" directly. That should probably be "pre-insert". And now I introduced "insert adjacent" it should probably reuse that: https://github.com/whatwg/dom/commit/8fa7ac749d1b612504184aad2c20808a1785b370.

**Comment by ArkadiuszMichalski** [comment]
["insert adjacent"](https://dom.spec.whatwg.org/#insert-adjacent) doesn't fit here because `insertAdjacentHTML()` for `"beforebegin"` and `"afterend"` throw `"NoModificationAllowedError"` when context doesn't have parent (Firefox/Chrome/IE11 do this).
For `insertAdjacentText()` and `insertAdjacentElement()` and the same position whe have:
Chrome/Firefox return null for both
IE11 throw for `insertAdjacentText()` but not for `insertAdjacentElement()` and return passing element.
Presto throw for `insertAdjacentElement()` but not for `insertAdjacentText()` and return undefined.
I will update tests in WPT to check this cases.

**Comment by annevk** [comment]
Hmm, https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/Source/core/dom/Element.cpp&q=insertadjacenthtml&sq=package:chromium&type=cs&l=2654 does show that some amount of code is shared. There might be an additional check needed though.

**Comment by ArkadiuszMichalski** [comment]
Right, probably work, just changing step 4. to use "insert adjacent", parent will be checking earlier.
